src/main/java/net/sf/mzmine/util/R/Rsession/Rdaemon.java

```diff
@@ -0,0 +1,285 @@
+package net.sf.mzmine.util.R.Rsession;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+import java.util.Properties;
+import net.sf.mzmine.util.R.Rsession.Logger.Level;
+import org.rosuda.REngine.Rserve.RConnection;
+
+/**
+ *
+ * @author richet
+ */
+public class Rdaemon {
+
+	RserverConf conf;
+	Process process;
+	private final Logger log;
+	static File APP_DIR = new File(System.getProperty("user.home") + File.separator + ".Rserve");
+	public static String R_HOME = null;
+
+	private static boolean RSERVE_INSTALLED = false;
+
+	static {
+		boolean app_dir_ok = false;
+		if (!APP_DIR.exists()) {
+			app_dir_ok = APP_DIR.mkdir();
+		} else {
+			app_dir_ok = APP_DIR.isDirectory() && APP_DIR.canWrite();
+		}
+		if (!app_dir_ok) {
+			System.err.println("Cannot write in " + APP_DIR.getAbsolutePath());
+		}
+	}
+
+	public Rdaemon(RserverConf conf, Logger log, String R_HOME) {
+		this.conf = conf;
+		this.log = log != null ? log : new Slf4jLogger();
+		findR_HOME(R_HOME);
+		log.println("Environment variables:\n  " + R_HOME_KEY + "=" + Rdaemon.R_HOME /*+ "\n  " + Rserve_HOME_KEY + "=" + Rdaemon.Rserve_HOME*/, Level.INFO);        
+		Runtime.getRuntime().addShutdownHook(new Thread() {
+			@Override
+			public void run() {
+				_stop();
+			}
+		});
+	}
+
+	private void _stop() {
+		stop();
+	}
+
+	public Rdaemon(RserverConf conf, Logger log) {
+		this(conf, log, null);
+	}
+	public final static String R_HOME_KEY = "R_HOME";
+
+	public static boolean findR_HOME(String r_HOME) {
+		Map<String, String> env = System.getenv();
+		Properties prop = System.getProperties();
+
+		if (r_HOME!=null) R_HOME = r_HOME;
+		if (R_HOME == null || !(new File(R_HOME).isDirectory())) {
+			if (env.containsKey(R_HOME_KEY)) {
+				R_HOME = env.get(R_HOME_KEY);
+			}
+
+			if (R_HOME == null || prop.containsKey(R_HOME_KEY) || !(new File(R_HOME).isDirectory())) {
+				R_HOME = prop.getProperty(R_HOME_KEY);
+			} 
+
+			if (R_HOME == null || !(new File(R_HOME).isDirectory())) {
+				R_HOME = "R";
+			}
+
+			if (R_HOME == null || !(new File(R_HOME).isDirectory())) {
+				R_HOME = null;
+				if (System.getProperty("os.name").contains("Win")) {
+					for (int major = 20; major >= 0; major--) {
+						//int major = 10;//known to work with R 2.9 only.
+						if (R_HOME == null) {
+							for (int minor = 10; minor >= 0; minor--) {
+								//int minor = 0;
+								r_HOME = "C:\\Program Files\\R\\R-3." + major + "." + minor + "\\";
+								if (new File(r_HOME).exists()) {
+									R_HOME = r_HOME;
+									break;
+								}
+							}
+						} else {
+							break;
+						}
+					}
+				} else {
+					R_HOME = "/usr/lib/R/";
+				}
+			}
+		}
+
+		if (R_HOME == null) {
+			return false;
+		}
+
+		return new File(R_HOME).isDirectory();
+	}
+
+	/*public static boolean findRserve_HOME(String path) {
```

Comment:
the findRserve_HOME function is not used? If it's not necessary, let's remove it.